### PR TITLE
Bg as search all internal api

### PR DESCRIPTION
**Pull request by bradgreen3**
Added internal API endpoints for searching users by whatever you want

### CODE REVIEW

File: spec/requests/api/v1/search_all_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,254 @@
+require 'rails_helper'
+
+RSpec.describe "General Search API" do
+  context  "GET api/v1/users/search" do
```
Comment (AliSchlereth) [inline spec/requests/api/v1/search_all_request_spec.rb]: This is an epic request spec.

File: app/models/user.rb [hunk 2]
```diff
@@ -99,4 +99,56 @@ def cohort_name
       "n/a"
     end
   end
+
+  def self.search_all(query)
```
Comment (AliSchlereth) [inline app/models/user.rb]: Great job extracting these out!

Comment (JStans12) [inline app/models/user.rb]: This is a pretty awesome method. Nice work team!

Comment (JStans12) [inline app/models/user.rb]: We should consider passing this logic along to something like `UserSearcher`.

**Pull request merged**